src/toil/batchSystems/slurm.py

```diff
@@ -461,10 +518,11 @@ def prepareSbatch(self,
                                 partition_name = option[len("--partition="):]
                             else:
                                 partition_name = option[i+1]
-
-                            if not any(partition_name == partition for partition in lowest_priority_gpus):
-                                # the specified partition is not compatible, so override with a gpu supported partition
-                                logger.warning(f"Job {jobName} needs {gpus} GPUs, but specified partition {partition_name} is incompatible. This job may not work.")
+                            available_gpu_partitions = self.boss.partitions.gpu_partitions
+                            if not any(partition_name == partition for partition in available_gpu_partitions):
```

Comment:
This will not work since the GPU partition collection is actually full of `PartitionInfo`s and not name strings. We should make it a set and test membership I think.

```suggestion
                           if partition_name not in available_gpu_partitions:
```